grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx

```diff
@@ -1,512 +1,632 @@
-    })();
-  }, []);
-
-  const data =
-    id === 'new'
-      ? { integration: selectedOption?.value, team: user?.current_team, labels: [] }
+const RADIO_OPTIONS = [
+  {
+    label: 'Connect existing Contact point',
+    value: 'existing',
+  },
+  {
+    label: 'Create a new one',
+    value: 'new',
+  },
+];
+
+export const IntegrationForm = observer(
+  ({
+    id,
+    isTableView,
+    navigateToAlertGroupLabels,
+    selectedIntegration,
+    onSubmit,
+    onHide,
+    onBackClick,
+  }: IntegrationFormProps) => {
+    const store = useStore();
+    const history = useHistory();
+    const styles = useStyles2(getIntegrationFormStyles);
+    const isNew = id === 'new';
+    const { userStore, grafanaTeamStore, alertReceiveChannelStore } = store;
+
+    const data = isNew
+      ? { integration: selectedIntegration?.value, team: userStore.currentUser?.current_team, labels: [] }
       : prepareForEdit(alertReceiveChannelStore.items[id]);
 
-  const { alertReceiveChannelOptions } = alertReceiveChannelStore;
-
-  const options = alertReceiveChannelOptions
-    ? alertReceiveChannelOptions.filter((option: ApiSchemas['AlertReceiveChannelIntegrationOptions']) => {
-        if (option.value === 'grafana_alerting' && !window.grafanaBootData.settings.unifiedAlertingEnabled) {
-          return false;
-        }
+    const { integration } = data;
+
+    const formMethods = useForm<FormFields>({
+      defaultValues: isNew
+        ? {
+            // these are the default values for creating an integration
+            [FormFieldKeys.Integration]: integration,
+            [FormFieldKeys.DefaultWebhooks]: true,
+          }
+        : {
+            // existing values from existing integration (edit-mode)
+            ...data,
+          },
+      mode: 'onChange',
+    });
+    const {
+      control,
+      handleSubmit,
+      formState: { errors },
+    } = formMethods;
+
+    const [
+      {
+        isExistingContactPoint,
+        dataSources,
+        contactPoints,
+        selectedAlertManagerOption,
+        selectedContactPointOption,
+        allContactPoints,
+      },
+      setState,
+    ] = useReducer(
+      (state: GrafanaContactPointState, newState: Partial<GrafanaContactPointState>) => ({
+        ...state,
+        ...newState,
+      }),
+      {
+        isExistingContactPoint: true,
+        selectedAlertManagerOption: undefined,
+        selectedContactPointOption: undefined,
+        dataSources: [],
+        contactPoints: [],
+        allContactPoints: [],
+      }
+    );
+
+    useEffect(() => {
+      (async function () {
+        setState({
+          allContactPoints: await AlertReceiveChannelHelper.getGrafanaAlertingContactPoints(),
+        });
+      })();
+    }, []);
+
+    const labelsRef = useRef(null);
+
+    const [labelsErrors, setLabelErrors] = useState([]);
+    const isServiceNow = getIsBidirectionalIntegration(data as Partial<ApiSchemas['AlertReceiveChannel']>);
```

Comment:
we can get rid of casting and improve typing as shown here https://github.com/grafana/oncall/compare/rares/integration-forms...brojd/integration-forms-type-fix?expand=1 

Can be done later though if you want

Reply:
Will merge this in a separate PR then